### PR TITLE
added support for months_between for NZ

### PR DESCRIPTION
**Pull request by rohitgaurDatametica**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -861,6 +861,16 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
               daySymbolLiteral, call.operand(0));
       super.unparseCall(writer, extractCall, leftPrec, rightPrec);
       break;
+    case "MONTHS_BETWEEN":
+      final SqlWriter.Frame dateDiffFrame = writer.startFunCall("DATE_DIFF");
+      writer.sep(",");
```
Comment (deepakDatametica) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:866]: don't use writer. Go with SqlCall approach.

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 2]
```diff
@@ -859,5 +860,14 @@ private SqlLibraryOperators() {
           ReturnTypes.VARCHAR_2000_NULLABLE, null,
               OperandTypes.family(SqlTypeFamily.NUMERIC, SqlTypeFamily.STRING),
           SqlFunctionCategory.STRING);
+
+  @LibraryOperator(libraries = {NETEZZA})
+  public static final SqlFunction MONTHS_BETWEEN =
```
Comment (deepakDatametica) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java:865]: Test cases are missing.

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 3]
```diff
@@ -862,14 +865,26 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
...
+      SqlNode monthSymbol = SqlLiteral.createSymbol(TimeUnit.MONTH, SqlParserPos.ZERO);
+      SqlNode dateDiffNode = DATE_DIFF.createCall(SqlParserPos.ZERO,
+              call.operand(0), call.operand(1), monthSymbol);
+
+      daySymbolLiteral = SqlLiteral.createSymbol(TimeUnit.DAY, SqlParserPos.ZERO);
+
+      SqlNode day1 = EXTRACT.createCall(SqlParserPos.ZERO, daySymbolLiteral, call.operand(0));
+      SqlNode day2 = EXTRACT.createCall(SqlParserPos.ZERO, daySymbolLiteral, call.operand(1));
+
+      SqlNode subtarctNode = MINUS.createCall(SqlParserPos.ZERO, day1, day2);
```
Comment (deepakDatametica) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:877]: variable typo mistake

File: core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java [hunk 4]
```diff
@@ -3130,6 +3130,28 @@ public static Object toCharFunction(Object value, Object format) {
   public static Object strTok(Object value, Object delimiter, Object part) {
     return ((String) value).split((String) delimiter) [(Integer) part - 1];
   }
+
+  public static Object monthsBetween(Object date1, Object date2) {
+    String[] dateSplit1 = ((String) date1).split("-");
```
Comment (vaibhavjain-dm) [inline core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java:3135]: give meaningful variable name.

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 5]
```diff
@@ -861,11 +864,37 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
...
   }
 
+  private void unparseMonthsBetween(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    SqlNode monthSymbol = SqlLiteral.createSymbol(TimeUnit.MONTH, SqlParserPos.ZERO);
+    SqlNode dateDiffNode = DATE_DIFF.createCall(SqlParserPos.ZERO,
+            call.operand(0), call.operand(1), monthSymbol);
+
+    SqlNode daySymbolLiteral = SqlLiteral.createSymbol(TimeUnit.DAY, SqlParserPos.ZERO);
+
+    SqlNode day1 = EXTRACT.createCall(SqlParserPos.ZERO, daySymbolLiteral, call.operand(0));
```
Comment (vaibhavjain-dm) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:882]: Give some meaningful name.

**Review by deepakDatametica — APPROVED** [review]

**Pull request merged**